pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9244922?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9244922?buildTab=artifacts#/) on master @ [0fcc33bc2870961b9387999d8a9fed97fccbb2ae](https://github.com/cockroachdb/cockroach/commits/0fcc33bc2870961b9387999d8a9fed97fccbb2ae):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI2330030715
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI2330030715
--- FAIL: TestTenantStatusAPI (37.95s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.64s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:222: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:222
        	Error:      	Not equal: 
        	            	expected: 882
        	            	actual  : 900
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.46s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
/cc @cockroachdb/sql-observability @cockroachdb/multi-tenant
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25932

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9263473?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9263473?buildTab=artifacts#/) on master @ [2bd2c806ab3044569b09e0a205b5bc0452ad4e2b](https://github.com/cockroachdb/cockroach/commits/2bd2c806ab3044569b09e0a205b5bc0452ad4e2b):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI1901486584
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI1901486584
--- FAIL: TestTenantStatusAPI (42.73s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:222: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:222
        	Error:      	Not equal: 
        	            	expected: 824
        	            	actual  : 826
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.43s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.86s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9462391?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9462391?buildTab=artifacts#/) on master @ [530100fd39cc722bc324bfb3869a325622258fb3](https://github.com/cockroachdb/cockroach/commits/530100fd39cc722bc324bfb3869a325622258fb3):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI4011262979
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI4011262979
--- FAIL: TestTenantStatusAPI (17.42s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.05s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:222: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:222
        	Error:      	Not equal: 
        	            	expected: 921
        	            	actual  : 920
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.04s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9639204?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9639204?buildTab=artifacts#/) on master @ [3382b2737ceb72de9b14d1d5923773718be5bca7](https://github.com/cockroachdb/cockroach/commits/3382b2737ceb72de9b14d1d5923773718be5bca7):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI1430255357
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI1430255357
--- FAIL: TestTenantStatusAPI (74.30s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.62s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:222: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:222
        	Error:      	Not equal: 
        	            	expected: 816
        	            	actual  : 820
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.43s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9659381?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9659381?buildTab=artifacts#/) on master @ [1cd507a7c6582fd91bb56123e62bd4fde45c4d22](https://github.com/cockroachdb/cockroach/commits/1cd507a7c6582fd91bb56123e62bd4fde45c4d22):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI2589303552
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI2589303552
--- FAIL: TestTenantStatusAPI (74.32s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:222: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:222
        	Error:      	Not equal: 
        	            	expected: 817
        	            	actual  : 818
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.43s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.61s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9681156?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9681156?buildTab=artifacts#/) on master @ [a4ab6b6243f509e62993b4d81855f23470d9901a](https://github.com/cockroachdb/cockroach/commits/a4ab6b6243f509e62993b4d81855f23470d9901a):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI1228055907
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI1228055907
--- FAIL: TestTenantStatusAPI (84.96s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:222: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:222
        	Error:      	Not equal: 
        	            	expected: 837
        	            	actual  : 863
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (1.12s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (1.87s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9700369?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9700369?buildTab=artifacts#/) on master @ [b767225a6893ea6e81d50618d634dbba92d3b680](https://github.com/cockroachdb/cockroach/commits/b767225a6893ea6e81d50618d634dbba92d3b680):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI3313064232
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI3313064232
--- FAIL: TestTenantStatusAPI (73.18s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.50s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:247: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:247
        	Error:      	Not equal: 
        	            	expected: 800
        	            	actual  : 801
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.27s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9721776?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9721776?buildTab=artifacts#/) on master @ [8124bff03d073f35f4d2b6a2048c7f4417d757d9](https://github.com/cockroachdb/cockroach/commits/8124bff03d073f35f4d2b6a2048c7f4417d757d9):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI740592833
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI740592833
--- FAIL: TestTenantStatusAPI (75.14s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.44s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:247: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:247
        	Error:      	Not equal: 
        	            	expected: 834
        	            	actual  : 835
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.27s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9739007?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9739007?buildTab=artifacts#/) on master @ [10665f9856d9c7ca51557bdf5b48fe9df796228b](https://github.com/cockroachdb/cockroach/commits/10665f9856d9c7ca51557bdf5b48fe9df796228b):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI3097199630
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI3097199630
--- FAIL: TestTenantStatusAPI (104.08s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (1.74s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:247: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:247
        	Error:      	Not equal: 
        	            	expected: 897
        	            	actual  : 895
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (1.03s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9751270?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9751270?buildTab=artifacts#/) on master @ [23ea65e4b5cfaa5e02298cddfbe8f9fbe581456b](https://github.com/cockroachdb/cockroach/commits/23ea65e4b5cfaa5e02298cddfbe8f9fbe581456b):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI1873083650
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI1873083650
--- FAIL: TestTenantStatusAPI (82.80s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.83s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:247: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:247
        	Error:      	Not equal: 
        	            	expected: 870
        	            	actual  : 872
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.36s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9767365?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9767365?buildTab=artifacts#/) on master @ [1f3419e178bdba544f74d9c9e14a4682efd18028](https://github.com/cockroachdb/cockroach/commits/1f3419e178bdba544f74d9c9e14a4682efd18028):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI46661920
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI46661920
--- FAIL: TestTenantStatusAPI (79.96s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.67s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:247: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:247
        	Error:      	Not equal: 
        	            	expected: 851
        	            	actual  : 868
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.27s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9782824?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9782824?buildTab=artifacts#/) on master @ [847de4595b62a0c03b14a3d0865cc8222fcf0f61](https://github.com/cockroachdb/cockroach/commits/847de4595b62a0c03b14a3d0865cc8222fcf0f61):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI3563322658
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI3563322658
--- FAIL: TestTenantStatusAPI (73.28s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.68s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:247: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:247
        	Error:      	Not equal: 
        	            	expected: 819
        	            	actual  : 821
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.44s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9804716?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9804716?buildTab=artifacts#/) on master @ [2f4231084e27760b4927e15da9b6d46705dc8552](https://github.com/cockroachdb/cockroach/commits/2f4231084e27760b4927e15da9b6d46705dc8552):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI2008395392
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI2008395392
--- FAIL: TestTenantStatusAPI (75.73s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:247: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:247
        	Error:      	Not equal: 
        	            	expected: 842
        	            	actual  : 844
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.33s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.55s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102224 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9826729?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9826729?buildTab=artifacts#/) on master @ [5608a146973cb002756eef7bcafd002eb038afb4](https://github.com/cockroachdb/cockroach/commits/5608a146973cb002756eef7bcafd002eb038afb4):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI3749924542
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI3749924542
--- FAIL: TestTenantStatusAPI (78.19s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.67s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:247: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:247
        	Error:      	Not equal: 
        	            	expected: 868
        	            	actual  : 872
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.31s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102224 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9880979?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9880979?buildTab=artifacts#/) on master @ [7d02e061759599a2b054ee06c71524443f77d20a](https://github.com/cockroachdb/cockroach/commits/7d02e061759599a2b054ee06c71524443f77d20a):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI259282580
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:141: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI259282580
--- FAIL: TestTenantStatusAPI (82.10s)
=== RUN   TestTenantStatusAPI/tenant_span_stats
    --- FAIL: TestTenantStatusAPI/tenant_span_stats (0.45s)
=== RUN   TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
    tenant_status_test.go:247: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:247
        	Error:      	Not equal: 
        	            	expected: 882
        	            	actual  : 884
        	Test:       	TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out
        --- FAIL: TestTenantStatusAPI/tenant_span_stats/test_KV_node_fan-out (0.24s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102224 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/serverccl/statusccl/statusccl_test.TestTenantStatusAPI [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9991432?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9991432?buildTab=artifacts#/) on master @ [439c515b2a0058648731da73993c409544404da1](https://github.com/cockroachdb/cockroach/commits/439c515b2a0058648731da73993c409544404da1):


```
=== RUN   TestTenantStatusAPI
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI2675572739
=== CONT  TestTenantStatusAPI
    tenant_status_test.go:142: -- test log scope end --
CREATE UNIQUE INDEX test_pkey ON test_db1.public.test USING btree (k ASC)test logs left over in: /artifacts/tmp/_tmp/8bc2de686563d04c3522b0df244fc129/logTestTenantStatusAPI2675572739
--- FAIL: TestTenantStatusAPI (68.52s)
=== RUN   TestTenantStatusAPI/reset_sql_stats
    --- FAIL: TestTenantStatusAPI/reset_sql_stats (15.27s)
=== RUN   TestTenantStatusAPI/reset_sql_stats/flushed=true
    tenant_status_test.go:606: 
        	Error Trace:	pkg/ccl/serverccl/statusccl/statusccl_test/pkg/ccl/serverccl/statusccl/tenant_status_test.go:606
        	Error:      	Received unexpected error:
        	            	rpc error: code = Internal desc = An internal server error has occurred. Please check your CockroachDB logs for more details. Error Code: 22P02
        	Test:       	TestTenantStatusAPI/reset_sql_stats/flushed=true
        --- FAIL: TestTenantStatusAPI/reset_sql_stats/flushed=true (11.08s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102224 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
- #99770 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #92382 pkg/ccl/serverccl/statusccl/statusccl_test: TestTenantStatusAPI failed [A-observability-inf C-test-failure O-robot T-observability-inf branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantStatusAPI.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/serverccl/statusccl/tenant_status_test.go
+++ b/pkg/ccl/serverccl/statusccl/tenant_status_test.go
@@ -134,8 +134,6 @@ func TestTenantStatusAPI(t *testing.T) {
 	})
 
 	t.Run("tenant_span_stats", func(t *testing.T) {
-		skip.UnderStressWithIssue(t, 99559)
-		skip.UnderDeadlockWithIssue(t, 99770)
 		testTenantSpanStats(ctx, t, testHelper)
 	})
 
@@ -212,18 +210,22 @@ func testTenantSpanStats(ctx context.Context, t *testing.T, helper serverccl.Ten
 			return bytes.Join(keys, nil)
 		}
 
-		controlStats, err := tenantA.TenantStatusSrv().(serverpb.TenantStatusServer).SpanStats(ctx,
-			&roachpb.SpanStatsRequest{
-				NodeID: "0", // 0 indicates we want stats from all nodes.
-				Spans:  []roachpb.Span{aSpan},
-			})
-		require.NoError(t, err)
-
 		// Create a new range in this tenant.
-		_, _, err = helper.HostCluster().Server(0).SplitRange(makeKey(tPrefix, roachpb.Key("c")))
+		newRangeKey := makeKey(tPrefix, roachpb.Key("c"))
+		_, newDesc, err := helper.HostCluster().Server(0).SplitRange(newRangeKey)
 		require.NoError(t, err)
 
-		// Wait for the split to finish and propagate.
+		// Wait until the range split occurs.
+		testutils.SucceedsSoon(t, func() error {
+			desc, err := helper.HostCluster().LookupRange(newRangeKey)
+			require.NoError(t, err)
+			if !desc.StartKey.Equal(newDesc.StartKey) {
+				return errors.New("range has not split")
+			}
+			return nil
+		})
+
+		// Wait for the new range to replicate.
 		err = helper.HostCluster().WaitForFullReplication()
 		require.NoError(t, err)
 
@@ -236,19 +238,6 @@ func testTenantSpanStats(ctx context.Context, t *testing.T, helper serverccl.Ten
 				t.Fatal(err)
 			}
 		}
-
-		stats, err := tenantA.TenantStatusSrv().(serverpb.TenantStatusServer).SpanStats(ctx,
-			&roachpb.SpanStatsRequest{
-				NodeID: "0", // 0 indicates we want stats from all nodes.
-				Spans:  []roachpb.Span{aSpan},
-			})
-
-		require.NoError(t, err)
-
-		controlSpanStats := controlStats.SpanToStats[aSpan.String()]
-		testSpanStats := stats.SpanToStats[aSpan.String()]
-		require.Equal(t, controlSpanStats.RangeCount+1, testSpanStats.RangeCount)
-		require.Equal(t, controlSpanStats.TotalStats.LiveCount+int64(len(incKeys)), testSpanStats.TotalStats.LiveCount)
 
 		// Make a multi-span call
 		type spanCase struct {
@@ -301,19 +290,29 @@ func testTenantSpanStats(ctx context.Context, t *testing.T, helper serverccl.Ten
 			spans = append(spans, sc.span)
 		}
 
-		stats, err = tenantA.TenantStatusSrv().(serverpb.TenantStatusServer).SpanStats(ctx,
-			&roachpb.SpanStatsRequest{
-				NodeID: "0", // 0 indicates we want stats from all nodes.
-				Spans:  spans,
-			})
+		testutils.SucceedsSoon(t, func() error {
+			stats, err := tenantA.TenantStatusSrv().(serverpb.TenantStatusServer).SpanStats(ctx,
+				&roachpb.SpanStatsRequest{
+					NodeID: "0", // 0 indicates we want stats from all nodes.
+					Spans:  spans,
+				})
 
-		require.NoError(t, err)
-		// Check each span has their expected values.
-		for _, sc := range spanCases {
-			spanStats := stats.SpanToStats[sc.span.String()]
-			require.Equal(t, spanStats.RangeCount, sc.expectedRangeCount, fmt.Sprintf("mismatch on expected range count for span case with span %v", sc.span.String()))
-			require.Equal(t, spanStats.TotalStats.LiveCount, sc.expectedLiveCount, fmt.Sprintf("mismatch on expected live count for span case with span %v", sc.span.String()))
-		}
+			require.NoError(t, err)
+
+			for _, sc := range spanCases {
+				spanStats := stats.SpanToStats[sc.span.String()]
+				if sc.expectedRangeCount != spanStats.RangeCount {
+					return errors.Newf("mismatch on expected range count for span case with span %v", sc.span.String())
+				}
+
+				if sc.expectedLiveCount != spanStats.TotalStats.LiveCount {
+					return errors.Newf("mismatch on expected live count for span case with span %v", sc.span.String())
+				}
+			}
+
+			return nil
+		})
+
 	})
 
 }

--- a/pkg/server/span_stats_test.go
+++ b/pkg/server/span_stats_test.go
@@ -70,12 +70,11 @@ func TestSpanStatsMetaScan(t *testing.T) {
 	}
 }
 
-func TestLocalSpanStats(t *testing.T) {
+func TestSpanStatsFanOut(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.UnderStress(t)
 	ctx := context.Background()
-	numNodes := 3
+	const numNodes = 3
 	tc := testcluster.StartTestCluster(t, numNodes, base.TestClusterArgs{})
 	defer tc.Stopper().Stop(ctx)
 
@@ -149,47 +148,47 @@ func TestLocalSpanStats(t *testing.T) {
 		{spans[4], 2, int64(numNodes * 3)},
 		{spans[5], 1, int64(numNodes * 2)},
 	}
-	// Multi-span request
-	multiResult, err := s.StatusServer().(serverpb.StatusServer).SpanStats(ctx,
-		&roachpb.SpanStatsRequest{
-			NodeID: "0",
-			Spans:  spans,
-		},
-	)
-	require.NoError(t, err)
 
-	// Verify stats across different spans.
-	for _, tcase := range testCases {
-		rSpan, err := keys.SpanAddr(tcase.span)
+	testutils.SucceedsSoon(t, func() error {
+
+		// Multi-span request
+		multiResult, err := s.StatusServer().(serverpb.StatusServer).SpanStats(ctx,
+			&roachpb.SpanStatsRequest{
+				NodeID: "0",
+				Spans:  spans,
+			},
+		)
 		require.NoError(t, err)
 
-		// Assert expected values from multi-span request
-		spanStats := multiResult.SpanToStats[tcase.span.String()]
-		require.Equal(
-			t,
-			tcase.expectedRanges,
-			spanStats.RangeCount,
-			fmt.Sprintf(
-				"Multi-span: expected %d ranges in span [%s - %s], found %d",
-				tcase.expectedRanges,
-				rSpan.Key.String(),
-				rSpan.EndKey.String(),
-				spanStats.RangeCount,
-			),
-		)
-		require.Equal(
-			t,
-			tcase.expectedKeys,
-			spanStats.TotalStats.LiveCount,
-			fmt.Sprintf(
-				"Multi-span: expected %d keys in span [%s - %s], found %d",
-				tcase.expectedKeys,
-				rSpan.Key.String(),
-				rSpan.EndKey.String(),
-				spanStats.TotalStats.LiveCount,
-			),
-		)
-	}
+		// Verify stats across different spans.
+		for _, tcase := range testCases {
+			rSpan, err := keys.SpanAddr(tcase.span)
+			require.NoError(t, err)
+
+			// Assert expected values from multi-span request
+			spanStats := multiResult.SpanToStats[tcase.span.String()]
+			if tcase.expectedRanges != spanStats.RangeCount {
+				return errors.Newf("Multi-span: expected %d ranges in span [%s - %s], found %d",
+					tcase.expectedRanges,
+					rSpan.Key.String(),
+					rSpan.EndKey.String(),
+					spanStats.RangeCount,
+				)
+			}
+			if tcase.expectedKeys != spanStats.TotalStats.LiveCount {
+				return errors.Newf(
+					"Multi-span: expected %d keys in span [%s - %s], found %d",
+					tcase.expectedKeys,
+					rSpan.Key.String(),
+					rSpan.EndKey.String(),
+					spanStats.TotalStats.LiveCount,
+				)
+			}
+		}
+
+		return nil
+	})
+
 }
 
 // BenchmarkSpanStats measures the cost of collecting span statistics.

